pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,762 @@
+	allocatedClaimWithGoodTopology = toResourceClaim(resourcev1alpha1ac.ResourceClaim(claimName, namespace).
+					WithOwnerReferences(metav1ac.OwnerReference().
+						WithUID(podUID).
+						WithController(true)).
+					WithStatus(resourcev1alpha1ac.ResourceClaimStatus().
+						WithAllocation(resourcev1alpha1ac.AllocationResult().
+							WithAvailableOnNodes(corev1ac.NodeSelector().
+								WithNodeSelectorTerms(corev1ac.NodeSelectorTerm().
+									WithMatchExpressions(corev1ac.NodeSelectorRequirement().
+										WithKey("nodename").
+										WithOperator(v1.NodeSelectorOpIn).
+										WithValues("worker")))))))
+	otherClaim = toResourceClaim(resourcev1alpha1ac.ResourceClaim("not-my-claim", namespace))
+
+	schedulingSelectedPotential = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+					WithOwnerReferences(metav1ac.OwnerReference().
+						WithAPIVersion("v1").
+						WithController(true).
+						WithKind("Pod").
+						WithName(podName).
+						WithUID(podUID)).
+					WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+						WithSelectedNode(workerNode.Name).
+						WithPotentialNodes(workerNode.Name)))
+	schedulingPotential = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+				WithOwnerReferences(metav1ac.OwnerReference().
+					WithAPIVersion("v1").
+					WithController(true).
+					WithKind("Pod").
+					WithName(podName).
+					WithUID(podUID)).
+				WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+					WithPotentialNodes(workerNode.Name)))
+	schedulingInfo = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+			WithOwnerReferences(metav1ac.OwnerReference().
+				WithAPIVersion("v1").
+				WithController(true).
+				WithKind("Pod").
+				WithName(podName).
+				WithUID(podUID)).
+			WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+				WithPotentialNodes(workerNode.Name)).
+			WithStatus(resourcev1alpha1ac.PodSchedulingStatus().
+				WithResourceClaims(resourcev1alpha1ac.ResourceClaimSchedulingStatus().
+					WithName(resourceName), resourcev1alpha1ac.ResourceClaimSchedulingStatus().
+					WithName(resourceName2))))
+)
+
+// result defines the expected outcome of some operation. It covers return
+// values, the plugin state and the state of the world (= objects).
+type result struct {
+	returnValue interface{}
+	status      *framework.Status
+
+	// changes contains corev1ac apply configuration instances that contain
+	// the expected changes to existing objects. This way, new fields and
+	// can be described concisely. Removal of fields is not needed.
+	changes []any
+
+	// added contains objects created by the operation.
+	added []metav1.Object
+
+	// removed contains objects deleted by the operation.
+	removed []metav1.Object
+}
+
+type operation int
+
+const (
+	prefilterOp operation = iota
+	filterOp
+	prescoreOp
+	reserveOp
+	unreserveOp
+	postbindOp
+)
+
+type perNodeResult map[string]result
+
+type want map[operation]any
+
+func TestPreFilter(t *testing.T) {
```

Comment:
> Can you try to make them more unit-like? Probably something closer to a test per Plugin extension point would be better.

The current style of walking through all extension points is how the volume binding plugin is getting tested. I started with that, then had the same thought as you and wanted to split it into unit tests per extension point. The name TestPreFilter is a left-over from that attempt. I'll fix that.

I did not do it in the end because I realized that it has several drawbacks:
- If each extension points gets its own Test* function with different test cases, then I would have had a lot of duplication of test case definitions (basically the starting conditions). Later, when adding a new scenario in one function there's a risk of forgetting to add it elsewhere, for example because it doesn't seem relevant although it is.
- If I keep one function with the test cases and define sub-tests (one per extension point), then the test case setup code has to know about internals of the plugin (primarily the state) instead of just testing the externally visible behavior.

The second approach may be viable, but I am leaning towards keeping it as it is.